bb_proxy/src/conn.rs

```diff
@@ -153,9 +155,20 @@ pub struct JsonPlayer {
   pub id:   String,
 }
 
+impl LoginInfo {
+  pub fn offline(name: String) -> Self {
```

Comment:
Pass a `&str` here, not a `String`. Passing `&str` won't allocate, whereas a `String` will allocate.

Reply:
done